### PR TITLE
document installer URL usage

### PR DESCRIPTION
**Pull request by wfeldt**
## Task

Document current state of URL handling in the installer.

- https://trello.com/c/CAytlL3W

Usage:

- https://github.com/yast/yast-packager/pull/624
- https://github.com/yast/yast-add-on/pull/127

### CODE REVIEW

**Comment by mvidner** [comment]
Discoverability: the new doc should be linked from
- the repo's README
- the new code that follows the spec

File: doc/url.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+All three backends have a collection of their own URL formats. Not so much for
+standardized URL schemes as `http` or `ftp` - but schemes referring to local
+media vary.
+
+Some standarization attempts have been made in the past. As a result Linuxrc
+supports also Zypp and (Auto)YaST formats. Note that Linuxrc always uses the Zypp format
+when passing URLs to YaST in `install.inf`.
+
+`FileFromUrl` has been recently extended to work also with Zypp formats. The
```
Comment (mvidner) [inline doc/url.md:41]: It wouid be nice to note the code stream where it is usable
 
```suggestion
`FileFromUrl` has been recently (SLE99-SP9, Leap 99.9) extended to work also with Zypp formats. The
```

**Comment by wfeldt** [comment]
That's meant as an YaST internal document describing the situation from a developer 's view. There's plenty of public docs describing the user side.

File: doc/url.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+if !Builtins.regexpmatch(url, "^relurl://")
+  Builtins.y2debug("Not a relative URL: %1", URL.HidePassword(url))
+  return url
+end
+```
+
+There are often hidden assumptions in these regexps (e.g. that `relurl` starts with at least two slashes)
+that might break things at some point.
+
+But there is a perfectly fine `URI` class in ruby that can do this better. For example:
```
Comment (mvidner) [inline doc/url.md:144]: BTW `URI` in Ruby 2.5 has a super annoying feature of failing to recognize strings with a trailing `\n` while stripping that `\n` from the exception message :anger: . 3.1 reports the `\n`.

Comment (wfeldt) [reply doc/url.md]: You might want to add your own experiences to this document later. But I really just want to get this initial version going.

File: doc/url.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+On top of the above three locations there is special handling for `relurl`
+and `repo` URLs throughout the Linuxrc and YaST code - mostly concerned constructing the absolute URL.
+
+`relurl` is a relative URL where the location it is relative to varies on
+context. It can be relative to the installation repo, relative to the
+AutoYaST profile, or relative to the main product in the context of add-on
+or product descriptions.
+
+`repo` is a URL that is always relative to the installation repo.
```
Comment (mvidner) [inline doc/url.md:29]:
```suggestion
`repo` is a URL that is always relative to the installation repository.
Only its *path* is used, relative to the installation repository path+URL.
```

**Review by mvidner — APPROVED** [review]
Thanks!

I've added the link from README and now it LGTM

**Pull request merged**